string node_name
string path
uint32 distance